### PR TITLE
en passant capture

### PR DESCRIPTION
**Pull request by cbohara**
en passant is a special pawn capture that allows a pawn to capture an unfriendly pawn even if it is not at the destination.

### CODE REVIEW

File: app/models/pawn.rb [hunk 1]
```diff
@@ -49,6 +49,38 @@ def valid_move?(x,y)
     return false
   end
 
+  # returns true if unfriendly pawn on square
+  def unfriendly_pawn?(x,y)
+    piece = self.game.find_piece(x,y)
+    return (piece.type == "Pawn") && ((piece.white? != white?) || (piece.black? != black?))
```
Comment (jacobwgillespie) [inline app/models/pawn.rb]: You actually only need `piece.white? != white?` (or `piece.black? != black?`) because of the following:


`self` | `piece` | `piece.black?` | `black?` | `piece.black? != black?`
--- | --- | --- | --- | ---
White | White | `false` | `false` | `true`
White | Black | `false` | `true` | `false`
Black | White | `true` | `false` | `false`
Black | Black | `true` | `true` | `true`

tl;dr - you only need one

**Comment by jacobwgillespie** [comment]
I'm not sure the move is implemented correctly - *en passant* allows a pawn to capture an opponent's pawn after the opponent's pawn makes its special two-forward initial move.  It prevents your opponent from moving two spaces forward to avoid their pawn being captured if they'd only moved one space forward:

board

If the opponent moves to the X, then the white pawn can capture the black pawn.  However they could attempt to "escape" by moving their pawn forward two:

board

*En passant* then allows the white pawn to move to the X, the space that the capture would have taken place, and capture the pawn anyways:

board

From as best as I can tell, this PR allows vertical captures and isn't checking for the exact conditions above, e.g. the opposing pawn must have made its special initial move "past" where you would have just captured it.

More details: https://en.wikipedia.org/wiki/En_passant

File: app/models/pawn.rb (hunk 1 above)
Comment (cbohara) [inline app/models/pawn.rb]: ahh ok gotcha

**Comment by cbohara** [comment]
Ahh ok gotcha I missed the bit around it having to happen _only_ if the opposing pawn made the 2 square jump.  Thank you I will fix that.

File: db/migrate/20160805153018_add_pawn_two_square_to_pieces.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddPawnTwoSquareToPieces < ActiveRecord::Migration
+  def change
+    add_column :pieces, :pawn_two_squares, :boolean
```
Comment (jacobwgillespie) [inline db/migrate/20160805153018_add_pawn_two_square_to_pieces.rb]: What does this column do?  I'm not seeing where it is actually set.

Comment (cbohara) [inline db/migrate/20160805153018_add_pawn_two_square_to_pieces.rb]: @jacobwgillespie that is a very good question!  not sure how I forgot to add that logic, but I added it in my latest commit

**Comment by jacobwgillespie** [comment]
Hey, where are the `pawn_move_to` and `pawn_standard_capture` methods used?  I can't seem to find where they're utilized (I'm probably just missing something obvious).

**Comment by cbohara** [comment]
My bad!  I've incorporated the pawn logic into the move_to! logic in the piece model.  And all the tests are still passing!
